src/libraries/Microsoft.PowerFx.Connectors/Public/ConnectorType.cs

```diff
@@ -114,12 +114,12 @@ public class ConnectorType : SupportsConnectorErrors
 
         internal ISwaggerSchema Schema { get; private set; } = null;
 
-        // Relationships to external tables
-        internal List<string> ExternalTables { get; set; }
+        // Relationships to external tables        
+#pragma warning disable CS0618 // Type or member is obsolete
+        internal IEnumerable<ConnectorRelationship> ExternalTables => _externalTables;
 
-        internal string RelationshipName { get; set; }
-
-        internal string ForeignKey { get; set; }
+        private List<ConnectorRelationship> _externalTables;
```

Comment:
>List<ConnectorRelationship> [](http://example.com/codeflow?start=16&length=27)

nit: IEnumerable<>
I know its annoying but immutable collections are much easier to maintain/debug.